src/aind_codeocean_api/codeocean.py

```diff
@@ -101,6 +104,83 @@ def search_data_assets(
 
         return response
 
+    def search_all_data_assets(
+        self,
+        sort_order: Optional[str] = None,
+        sort_field: Optional[str] = None,
+        type: Optional[str] = None,
+        ownership: Optional[str] = None,
+        favorite: Optional[bool] = None,
+        archived: Optional[bool] = None,
+        query: Optional[str] = None,
+    ) -> requests.models.Response:
+        """
+        Utility method to return all the search results that match a query
+        Parameters
+        ----------
+        sort_order : Optional[str]
+            Determines the result sort order.
+        sort_field : Optional[str]
+            Determines the field to sort by.
+        type : Optional[str]
+            Type of data asset: dataset or result.
+            Returns both if omitted.
+        ownership : Optional[str]
+            Search data asset by ownership: owner or shared.
+        favorite : Optional[bool]
+            Search only favorite data assets.
+        archived : Optional[bool]
+            Search only archived data assets.
+        query : Optional[str]
+            Determines the search query.
+
+        Returns
+        -------
+        requests.models.Response
+        """
+
+        # TODO: it'd be nice to re-use the search_data_assets function, but
+        #  it'll require passing in a requests.Session object into that method.
+        optional_params = locals()
+        query_params = {}
+
+        for param, val in optional_params.items():
+            if val is not None:
+                query_params[param] = val
+
+        start_index = 0
+        has_more_results = True
+        requests_session = requests.Session()
+        start_index = 0
+        limit = self._MAX_SEARCH_BATCH_REQUEST
+        all_results = []
```

Comment:
were we going to `yield` instead of blocking?

Reply:
It depends on what object we want to return to the user. I wrap the results in a Response object to be consistent with the return objects of the other api calls, but I can make it so it returns an Iterable of results.